pkg/config/initoptions.go

```diff
@@ -142,19 +144,24 @@ func calcPackageNamespace(packageDir string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	// Return the first non-empty namespace found. Cluster-scoped
+	// Return the non-empty unique namespace if found. Cluster-scoped
 	// resources do not have namespace set.
+	currentNamespace := metav1.NamespaceDefault
 	for _, node := range nodes {
 		rm, err := node.GetMeta()
-		if err != nil {
+		if err != nil || len(rm.ObjectMeta.Namespace) <= 0 {
```

Comment:
We should not just ignore resources that doesn't have the namespace set. This means the resource is either cluster-scoped or namespace-scoped without a namespace set. In both cases I think we should probably ask the user to provide the correct namespace.

Reply:
I thought cluster scoped resources doesn't have namespace, thus won't make sense to ask user to provide. Or am I missing something here?

For the current setting, namespace scoped resource that doesn't have namespace set will be drop in the function getNamespacedResource.

Reply:
return error in this case